Fix models not displaying

JosselinSomervilleRoberts opened the pull request:
Fix #1615 .

I also added the model access and a warning message if the model is still in `todo`.

percyliang commented on docs/mkdocs_macros.py:
```diff
@@ -2,14 +2,32 @@
 def define_env(env):
     @env.macro
     def models_by_organization():
+        with open("src/helm/benchmark/static/schema.yaml", "r") as f:
```
Use `read_schema` to use the typed objects, which has the `name_to_model` field already...

percyliang commented on docs/mkdocs_macros.py:
```diff
@@ -2,14 +2,32 @@
+        for model in models:
+            # Find the model in ALL_MODELS in an efficient way
+            model_object = name_to_model_object.get(model["name"], None)
+            if model_object is not None:
```
In general, it'd be good to avoid using raw hash lookups and use dataclasses instead.

JosselinSomervilleRoberts commented:
I think this should be good now.

percyliang reviewed (commented):
Zooming out, there are two objects:
1) `ModelField` from `schema.yaml`: this has the nice descriptions, etc. and also includes models that we don't have implementations for.
2) `Model`: this is tied to implementations and has things like tags.

For the former, we already have documentation on the crfm.stanford.edu/helm page.  So I'm not sure what the purpose of `models.md` is?

JosselinSomervilleRoberts commented:
@percyliang, I might have understood your answer wrong but I think the page you are mentioning is our current page. It was broken, so I fixed it and added a few elements to it while I was at it.

percyliang commented:
I think the frontpage (https://crfm.stanford.edu/helm/latest/) is driven by Javascript, not the `models.md`.

JosselinSomervilleRoberts commented:
Following a discussion we had offline with @yifanmai , we are now redirecting to HELM's model page. Should I remove `models_by_organization()` as I think it is not being used anymore or keep it as it might become useful again in the future?

yifanmai commented on mkdocs.yml:
```diff
@@ -28,6 +28,7 @@ extra_css:
   - docstrings.css
 markdown_extensions:
   - pymdownx.magiclink
+  - pymdownx.extra
```
Don't think we need this any more?

JosselinSomervilleRoberts replied:
I can remove it yes

yifanmai reviewed (changes requested):
Do we still need the other changes if we're just redirecting to the page?

JosselinSomervilleRoberts commented:
Right now we are just redirecting, I was just thinking the changes in case you needed later, I can just delete the function also (but should not keep the old version as it was not working)
